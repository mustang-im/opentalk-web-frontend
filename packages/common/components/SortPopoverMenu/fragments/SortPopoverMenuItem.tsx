// SPDX-FileCopyrightText: OpenTalk GmbH <mail@opentalk.eu>
//
// SPDX-License-Identifier: EUPL-1.2
import { MenuItemUnstyledProps } from '@mui/base';
import { MenuItem, styled, Typography, SvgIcon } from '@mui/material';
import React from 'react';
import { useTranslation } from 'react-i18next';

import { DoneIcon } from '../../../assets/icons';

const ContainerMenuItem = styled(MenuItem)({
  justifyContent: 'space-between',
  '& .MuiTypography-root': {
    fontWeight: 400,
  },
});

const DoneIconStyledWrapper = styled(SvgIcon, {
  shouldForwardProp: (prop) => prop !== 'isActive',
})<{ isActive: boolean | undefined }>(({ theme, isActive }) => ({
  marginLeft: theme.spacing(1),
  '& svg': { fill: isActive ? theme.palette.primary.light : 'transparent' },
}));

interface SortPopoverMenuItemProps extends Omit<MenuItemUnstyledProps, 'onSelect'> {
  i18nKey: string;
  selected?: boolean;
  value: string;
  onSelect(value: string): void;
}

const SortPopoverMenuItem = ({ i18nKey, selected, onSelect, value, ...props }: SortPopoverMenuItemProps) => {
  const { t } = useTranslation();

  const onClickHandler = () => {
    onSelect(value);
  };

  return (
    <ContainerMenuItem {...props} onClick={onClickHandler}>
      <Typography>{t(i18nKey)}</Typography>
      <DoneIconStyledWrapper fontSize="inherit" isActive={selected}>
        <DoneIcon />
      </DoneIconStyledWrapper>
    </ContainerMenuItem>
  );
};

export default SortPopoverMenuItem;
