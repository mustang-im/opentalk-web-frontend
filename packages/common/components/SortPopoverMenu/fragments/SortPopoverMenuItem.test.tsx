// SPDX-FileCopyrightText: OpenTalk GmbH <mail@opentalk.eu>
//
// SPDX-License-Identifier: EUPL-1.2
import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';

/**
 * Mocking react-i18next module as we don't care about actual
 * text in the snapshots, we only want to be sure that used key
 * is not going to change.
 */
import SortPopoverMenuItem from './SortPopoverMenuItem';

jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str: string) => str,
    };
  },
  initReactI18next: {
    type: '3rdParty',
  },
}));

describe('<SortPopoverMenuItem />', () => {
  it('should render with required properties.', () => {
    expect(
      render(<SortPopoverMenuItem i18nKey="test-key" value="test-value" onSelect={jest.fn()} />)
    ).toMatchSnapshot();
  });

  it('should execute onSelect callback with value when clicked.', () => {
    const callback = jest.fn();
    render(<SortPopoverMenuItem i18nKey="test-key" value="test-value" onSelect={callback} selected={false} />);
    const li = screen.getByRole('menuitem');
    fireEvent.click(li);
    expect(callback).toBeCalledWith('test-value');
  });
});
