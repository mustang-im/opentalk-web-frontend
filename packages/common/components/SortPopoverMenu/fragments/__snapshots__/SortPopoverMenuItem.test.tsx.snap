// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortPopoverMenuItem /> should render with required properties. 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters css-4f940q-MuiButtonBase-root-MuiMenuItem-root"
        role="menuitem"
        tabindex="-1"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          test-key
        </p>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1f9du2t-MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters css-4f940q-MuiButtonBase-root-MuiMenuItem-root"
      role="menuitem"
      tabindex="-1"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        test-key
      </p>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1f9du2t-MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          focusable="false"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
